tests/test_proxy_connect.py

```diff
@@ -1,120 +0,0 @@
-import json
-import os
-import time
-
-from six.moves.urllib.parse import urlsplit, urlunsplit
-from threading import Thread
-from libmproxy import controller, proxy
-from netlib import http_auth
-from testfixtures import LogCapture
-
-from twisted.internet import defer
-from twisted.trial.unittest import TestCase
-from scrapy.utils.test import get_crawler
-from scrapy.http import Request
-from tests.spiders import SimpleSpider, SingleRequestSpider
-from tests.mockserver import MockServer
-
-
-class HTTPSProxy(controller.Master, Thread):
-
-    def __init__(self):
-        password_manager = http_auth.PassManSingleUser('scrapy', 'scrapy')
-        authenticator = http_auth.BasicProxyAuth(password_manager, "mitmproxy")
-        cert_path = os.path.join(os.path.abspath(os.path.dirname(__file__)),
-            'keys', 'mitmproxy-ca.pem')
-        server = proxy.ProxyServer(proxy.ProxyConfig(
-            authenticator = authenticator,
-            cacert = cert_path),
-            0)
-        self.server = server
-        Thread.__init__(self)
-        controller.Master.__init__(self, server)
-
-    def http_address(self):
-        return 'http://scrapy:scrapy@%s:%d' % self.server.socket.getsockname()
-
-
-def _wrong_credentials(proxy_url):
-    bad_auth_proxy = list(urlsplit(proxy_url))
-    bad_auth_proxy[1] = bad_auth_proxy[1].replace('scrapy:scrapy@', 'wrong:wronger@')
-    return urlunsplit(bad_auth_proxy)
-
-class ProxyConnectTestCase(TestCase):
```

Comment:
Are these tests removed because mitmproxy version we're using is Python 2-only? If so, it seems we need to upgrade mitmproxy, but keep tests.

Reply:
Great question.

We currently have `mitmproxy==0.10.1` and `netlib==0.10.1` in the test requirements, the commit message says "pin mitmproxy 0.10.1 as >0.11 does not work with tests". I also found #2545. I'll look into updating the test for the current mitmproxy.